### ISSUE TITLE
Screen reader appears to check checkbox when it doesn't

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/gravity-force-lab#257]
**Test device**
Dell
**Operating System**
Win 10
**Browser**
Firefox
**Problem description**
For https://github.com/phetsims/QA/issues/480. Found with NVDA.
When going through the scene description you can hit a place where the checkbox for constant size looks highlighted and the screen reader reads out the beginning of the section. However, you need to be on the next item to actually be over the checkbox. If you try to select it when it is not over the box the sim will make the selected sound, even though it hasn't been. This may cause some confusion, so it may need a quick look.

**Visuals**
https://drive.google.com/file/d/1UGnoM51H0ylA0jOqvg7FELBX_zPPgQ8h/view?usp=sharing

Troubleshooting information:

<details>

!!!!! DO NOT EDIT !!!!!
Name: ‪Gravity Force Lab‬
URL: https://phet-dev.colorado.edu/html/gravity-force-lab/2.2.0-rc.2/phet/gravity-force-lab_all_phet.html
Version: 2.2.0-rc.2 2020-02-13 23:11:52 UTC
Features missing: touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:73.0) Gecko/20100101 Firefox/73.0
Language: en-US
Window: 1536x750
Pixel Ratio: 2.5/1
WebGL: WebGL 1.0
GLSL: WebGL GLSL ES 1.0
Vendor: Mozilla (Mozilla)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 32767x32767
OES_texture_float: true
Dependencies JSON: {}

</details>

**Comment by zepumph** [comment]
I can reproduce this very easily.
* Go to firefox
* turn on NVDA
* navigate to the checkbox
* press up arrow once. This moves the virtual cursor to the label from the input (I think)
* The NVDA says "clickable not checked", and when you press enter/space, it doesn't select the input. 

@jessegreenberg can you speak to this issue? Is this how screen readers work, so this behavior is expected, or should you be able to actvate when having focus on the label?

**Comment by zepumph** [comment]
I could not reproduce this with the following two checkboxes, one is a simple case, and the other is copied from the PDOM:

```html
<html>
<p>this is fluff content</p>
<label for="hi">My checkbox</label>

<input id="hi" onclick="" type="checkbox"/>
<p>this is fluff content</p>
<p>more flufffff</p>
<label data-trail-id="671-672-769-775-771-1025-1024-1017" id="label-671-672-769-775-771-1025-1024-1017"
       data-focusable="false" class="a11y-sibling" for="671-672-769-775-771-1025-1024-1017" tabindex="-1">Constant
  Size</label>
<input data-trail-id="671-672-769-775-771-1025-1024-1017" id="671-672-769-775-771-1025-1024-1017" data-focusable="true"
       class="a11y-sibling" type="checkbox" onclick="" aria-disabled="false" tabindex="-1">
<p>more flufffff</p>
</html>

**Comment by zepumph** [comment]
>@zepumph, do you need aria-disabled="false"? This checkbox does not get enabled and disabled, it's just there, right?

That's right. This isn't getting enabled or disabled in this context.

**Comment by jessegreenberg** [comment]
Notes from some investigation with @zepumph today:

We are seeing the following DOM events when printing events reaching `display.accessibleDOMELement`
![image](https://user-images.githubusercontent.com/6396244/75707622-335b7f80-5c8d-11ea-98b2-9f464af79437.png)

So we can see that the checkbox is getting two `click` events, but we don't know why. We also see some strange `focus` related events when clicked with alternative input. This makes us think that possibly AccessibilityTree.js operations are causing this? We will start there next time.

**Comment by zepumph** [comment]
No AccessibilityTree operations are occurring.

**Comment by jessegreenberg** [comment]
Stepping through the events, I am seeing that they *do* have different targets. It seems that data-trail-id does only point to the primary sibling.

![image](https://user-images.githubusercontent.com/6396244/75716548-6ad22800-5c9d-11ea-82fc-d105a77147f7.png)

So we are getting a click event on the label, dispatching it through the scene graph. Then getting a click event on the input and dispatching through the scene graph again.

**Comment by zepumph** [comment]
This is working great! Thanks @jessegreenberg for getting to the bottom of things. See https://github.com/phetsims/a11y-research/issues/156 for third party bug report.

**Comment by zepumph** [comment]
Mirror inputs wrapper helped find a bug in the above commit. Sometimes event.target is null, and that's ok. I committed a fix above to master and the GFL rc branch.

**Comment by KatieWoe** [comment]
Saw something similar with Jaws, where it seemed to read through and list "bullet" first before the radio button when using keyboard nav, and clicking at the wrong time didn't make the change. Just scrolling with the arrows wouldn't make the change as in normal keyboard nav. Is this acceptable @zepumph?

**Comment by jessegreenberg** [comment]
> and clicking at the wrong time didn't make the change.

Thanks, I am not sure yet if this is an issue, I will take a look.

> Just scrolling with the arrows wouldn't make the change as in normal keyboard nav. Is this acceptable

This surprised us too and we submitted a bug report to Freedom Scientific in https://github.com/phetsims/a11y-research/issues/54. According to JAWS the behavior you described is expected.

**Comment by zepumph** [comment]
@jessegreenberg are you saying that the first item may be a bug, but the second one definitely isn't?

**Comment by jessegreenberg** [comment]
That is correct. But I just tested the follwoing HTML with JAWS
<details>

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>TEST PAGE</title>
  </head>
  <body>

  <ul>
    <li>
      <label for="radio1">Decimals</label>
      <input id="radio1" type="radio">
    </li>
    <li>
      <label for="radio2">Scientific Notation</label>
      <input id="radio2" type="radio"> 
    </li>
    <li>
      <label for="radio3">Hidden</label>
      <input id="radio3" type="radio"> 
    </li>

  </ul>

  </body>
</html>
```
</details>
And confirmed that it behaves exactly as @KatieWoe describes. So both items in https://github.com/phetsims/gravity-force-lab/issues/257#issuecomment-594675503 are the expected behavior for JAWS.

**Comment by zepumph** [comment]
Alright great to know. I'm going to close this issue then. Thanks team!
